From e0a8c0663debeb222bf78b97678f60929313b60a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 6 May 2020 07:24:05 +0200
Subject: [PATCH] Remove completions unsuitable for autoloading

The override of 'python*' completions, as well as the attempt
to './manage.py' completion are not going to work with autoloader.
Strip them.
---
 extras/django_bash_completion | 33 +--------------------------------
 1 file changed, 1 insertion(+), 32 deletions(-)

diff --git a/extras/django_bash_completion b/extras/django_bash_completion
index fa77d59aff..dfeb439a2e 100755
--- a/extras/django_bash_completion
+++ b/extras/django_bash_completion
@@ -37,35 +37,4 @@ _django_completion()
                    DJANGO_AUTO_COMPLETE=1 $1 ) )
 }
 # When the django-admin.py deprecation ends, remove django-admin.py.
-complete -F _django_completion -o default django-admin.py manage.py django-admin
-
-_python_django_completion()
-{
-    if [[ ${COMP_CWORD} -ge 2 ]]; then
-        local PYTHON_EXE=${COMP_WORDS[0]##*/}
-        if echo "$PYTHON_EXE" | grep -qE "python([3-9]\.[0-9])?"; then
-            local PYTHON_SCRIPT=${COMP_WORDS[1]##*/}
-            if echo "$PYTHON_SCRIPT" | grep -qE "manage\.py|django-admin(\.py)?"; then
-                COMPREPLY=( $( COMP_WORDS=( "${COMP_WORDS[*]:1}" )
-                               COMP_CWORD=$(( COMP_CWORD-1 ))
-                               DJANGO_AUTO_COMPLETE=1 ${COMP_WORDS[*]} ) )
-            fi
-        fi
-    fi
-}
-
-# Support for multiple interpreters.
-unset pythons
-if command -v whereis &>/dev/null; then
-    python_interpreters=$(whereis python | cut -d " " -f 2-)
-    for python in $python_interpreters; do
-        [[ $python != *-config ]] && pythons="${pythons} ${python##*/}"
-    done
-    unset python_interpreters
-    pythons=$(echo "$pythons" | tr " " "\n" | sort -u | tr "\n" " ")
-else
-    pythons=python
-fi
-
-complete -F _python_django_completion -o default $pythons
-unset pythons
+complete -F _django_completion -o default django-admin.py django-admin
-- 
2.28.0

